app/models/concerns/bulkrax/dynamic_record_lookup.rb

```diff
@@ -12,15 +12,15 @@ def find_record(identifier, importer_run_id = nil)
       # check for our entry in our current importer first
       importer_id = ImporterRun.find(importer_run_id).importer_id
       default_scope = { identifier: identifier, importerexporter_type: 'Bulkrax::Importer' }
-      record = Entry.find_by(default_scope.merge({ importerexporter_id: importer_id })) || Entry.find_by(default_scope)
 
-      # TODO(alishaevn): discuss whether we are only looking for Collection models here
-      # use ActiveFedora::Base.find(identifier) instead?
-      record ||= ::Collection.where(id: identifier).first # rubocop:disable Rails/FindBy
-      if record.blank?
-        available_work_types.each do |work_type|
-          record ||= work_type.where(id: identifier).first # rubocop:disable Rails/FindBy
-        end
+      begin
+        # the identifier parameter can be a :source_identifier or the id of an object
+        record = Entry.find_by(default_scope.merge({ importerexporter_id: importer_id })) || Entry.find_by(default_scope)
+        record ||= ActiveFedora::Base.find(identifier)
+      rescue NameError # if ActiveFedora isn't installed
+        record = nil
+      rescue ActiveFedora::ObjectNotFoundError
+        record = nil
```

Comment:
```suggestion
      # NameError for if ActiveFedora isn't installed
      rescue NameError, ActiveFedora::ObjectNotFoundError
        record = nil
```

I believe you can do this so you don't need two rescues doing the same thing

Reply:
ahh. I wasn't sure about chaining in rescue's. thanks!